disperser/dataapi/v2/reservation_collector.go

```diff
@@ -0,0 +1,109 @@
+package v2
+
+import (
+	"context"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/disperser/dataapi"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// ReservationExpirationCollector is a custom Prometheus collector that queries reservation data
+// and exposes metrics about expiring reservations
+type ReservationExpirationCollector struct {
+	subgraphClient dataapi.SubgraphClient
+	logger         logging.Logger
+
+	// Metrics
+	reservationsActive         prometheus.Gauge
+	reservationTimeUntilExpiry *prometheus.GaugeVec
+}
+
+// NewReservationExpirationCollector creates a new collector
+func NewReservationExpirationCollector(subgraphClient dataapi.SubgraphClient, metrics *dataapi.Metrics, logger logging.Logger) *ReservationExpirationCollector {
+	return &ReservationExpirationCollector{
+		subgraphClient: subgraphClient,
+		logger:         logger,
+		reservationsActive: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "eigenda_reservations_active",
+			Help: "Number of active reservations",
+		}),
+		reservationTimeUntilExpiry: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "eigenda_reservation_time_until_expiry_seconds",
+			Help: "Time until reservation expiration in seconds",
+		}, []string{"account"}),
+	}
+}
+
+// Describe implements prometheus.Collector
+func (c *ReservationExpirationCollector) Describe(ch chan<- *prometheus.Desc) {
+	c.reservationsActive.Describe(ch)
+	c.reservationTimeUntilExpiry.Describe(ch)
+}
+
+// Collect implements prometheus.Collector
+func (c *ReservationExpirationCollector) Collect(ch chan<- prometheus.Metric) {
+	// Update counts with timeout to prevent blocking Prometheus scrapes
+	ctx, cancel := context.WithTimeout(context.Background(), 8*time.Second)
+	defer cancel()
+
+	c.updateCounts(ctx)
+
+	// Collect metrics
+	c.reservationsActive.Collect(ch)
+	c.reservationTimeUntilExpiry.Collect(ch)
+}
+
+// updateCounts queries the GraphQL endpoint and updates the metrics
+func (c *ReservationExpirationCollector) updateCounts(ctx context.Context) {
```

Comment:
why `updateCounts` and not `updateMetrics`? it updates timeUntilExpiry which is not a "count"

Reply:
65c828af1d633f531075c9bea7e13dd3b24c1e42